leases/app/controllers/MediaLeaseController.scala

```diff
@@ -37,17 +37,38 @@ class MediaLeaseController(auth: Authentication, store: LeaseStore, config: Leas
     store.delete(id).map { _ => notifications.sendRemoveLease(lease.mediaId, id)}
   }
 
-  private def clearLeases(id: String) = store.getForMedia(id)
+  private def clearLeases(id: String) = Future.sequence(store.getForMedia(id)
     .flatMap(_.id)
-    .map(clearLease)
+    .flatten(clearLease))
 
   private def badRequest(e:  Seq[(JsPath, Seq[JsonValidationError])]) =
     respondError(BadRequest, "media-leases-parse-failed", JsError.toJson(e).toString)
 
+  private def prepareLeaseForSave(mediaLease: MediaLease, userId: Option[String]): MediaLease =
+    mediaLease.prepareForSave.copy(id = Some(UUID.randomUUID().toString), leasedBy = userId)
+
   private def addLease(mediaLease: MediaLease, userId: Option[String]) = {
-    val lease = mediaLease.prepareForSave.copy(id = Some(UUID.randomUUID().toString), leasedBy = userId)
-    store.put(lease).map { _ =>
-      notifications.sendAddLease(lease)
+    val lease = prepareLeaseForSave(mediaLease, userId)
+    lease.isSyndication match {
+      case true =>
+        val mediaId = mediaLease.mediaId
```

Comment:
minor: we could use a for comprehension here to be ever so slightly more readable, but only just.